library/core/src/main/java/com/google/android/exoplayer2/mediacodec/MediaCodecInfo.java

```diff
@@ -624,6 +624,10 @@ private static int adjustMaxInputChannelCount(String name, String mimeType, int
         || MimeTypes.AUDIO_FLAC.equals(mimeType)
         || MimeTypes.AUDIO_ALAW.equals(mimeType)
         || MimeTypes.AUDIO_MLAW.equals(mimeType)
+        || MimeTypes.AUDIO_DTS.equals(mimeType)
+        || MimeTypes.AUDIO_DTS_HD.equals(mimeType)
+        || MimeTypes.AUDIO_DTS_EXPRESS.equals(mimeType)
```

Comment:
I don't think these should be added here (see https://github.com/google/ExoPlayer/issues/2940#issuecomment-308106108 for more details). Is there a specific reason to add them here?

Reply:
I think it should, forgive me if I'm wrong but if it wasn't there it would fall-back to an assumedMaxChannelCount of 30, rather than the maxChannelCount being passed in from the audioCapabilities.

Reply:
OK I see what's going on here, I've undone that change.